src/System.IO.FileSystem/tests/Directory/SetCreationTime_str_dt.cs

```diff
@@ -3,300 +3,34 @@
-                    iCountErrors++;
-                    printerr("Error_0022! Creation time cannot be correct");
-                }
-            }
-            catch (Exception exc)
-            {
-                iCountErrors++;
-                printerr("Error_0023! Unexpected exception thrown: " + exc.ToString());
-            }
-            EnsureDirectoryDeletion(directoryName);
-
-            //Subtract one day from DateTime.today.
-            strLoc = "Loc_0024";
-
-            EnsureDirectoryCreation(directoryName);
-            iCountTestcases++;
-            try
-            {
-                Directory.SetCreationTime(directoryName, DateTime.Now.AddDays(-1));
-                if ((Directory.GetCreationTime(directoryName) - DateTime.Now.AddDays(-1)).Seconds > 0)
-                {
-                    iCountErrors++;
-                    printerr("Error_0025! Creation time cannot be correct");
-                }
-            }
-            catch (Exception exc)
-            {
-                iCountErrors++;
-                printerr("Error_0026! Unexpected exception thrown: " + exc.ToString());
-            }
-            EnsureDirectoryDeletion(directoryName);
-
-            //With invalid datetime object.
-            strLoc = "Loc_0025";
-
-            EnsureDirectoryCreation(directoryName);
-            iCountTestcases++;
-            try
-            {
-                Directory.SetCreationTime(directoryName, new DateTime(2001, 332, 20, 50, 50, 50));
-                iCountErrors++;
-                printerr("Error_0026! Creation time cannot be correct");
-            }
-            catch (ArgumentOutOfRangeException)
-            {
-            }
-            catch (Exception exc)
-            {
-                iCountErrors++;
-                printerr("Error_0027! Unexpected exception thrown: " + exc.ToString());
-            }
-            EnsureDirectoryDeletion(directoryName);
-
-            //With valid date and time. 
-            strLoc = "Loc_0028";
-
-            EnsureDirectoryCreation(directoryName);
-            iCountTestcases++;
-            try
-            {
-                DateTime dt = new DateTime(2001, 2, 2, 20, 20, 20);
-                Directory.SetCreationTime(directoryName, dt);
-                if ((Directory.GetCreationTime(directoryName) - dt).Seconds > 0)
-                {
-                    iCountErrors++;
-                    printerr("Error_0029! Creation time cannot be correct");
-                }
-            }
-            catch (Exception exc)
-            {
-                iCountErrors++;
-                printerr("Error_0030! Unexpected exception thrown: " + exc.ToString());
-            }
-            EnsureDirectoryDeletion(directoryName);
+            return Directory.GetCreationTime(path);
         }
-        catch (Exception exc_general)
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows | PlatformID.OSX)] // birth time only available on some Unix systems
```

Comment:
Setting creation time is not supported on OSX; any tests that try to do so should just be [PlatformSpecific(PlatformID.Windows)].  Getting creation time is supported on OSX (but still not on Linux).